### ISSUE TITLE
Xero capabilities and limitations - 1 day spike

### ISSUE THREAD

**Issue by iteles** [dwyl/hq#383]
As discussed on 26-Jun-2017 and captured in https://github.com/dwyl/hq/issues/321#issuecomment-323071446, we need a spike to better understand how we can utilise Xero to its full potential and in a way that benefits our needs.

Including (but not limited to) understanding how to:
+ [ ] Add contractor invoices to the application in as automated a fashion as possible (i.e. is there a way which would avoid manual entry of invoices emailed to us)
  + This must allow us to track the expense again the invoice raised to the client or against that specific project as a whole #373 
+ [ ] Pull information out for year end accounts - I suspect Xero  offers an HMRC-ready format of some kind, but needs to be provided with the correct information https://github.com/dwyl/hq/issues/321#issuecomment-323071446
+ [ ] Address [these pain points](https://github.com/dwyl/hq/issues/382)
+ [ ] Add expenses in a foreign currency
+ [ ] Adding categorisations for expenses in a format that eases the burden for year end accounts #339 
+ [ ] Bulk upload expenses and in what format these need to be
+ [ ] Correctly get bank feeds to update #236

**Comment by markwilliamfirth** [comment]
All of these issues will take much longer than a day - or do you mean just understanding the process of how to solve each one?

**Comment by iteles** [comment]
> we need a spike to better understand how we can utilise Xero to its full potential 

Yes, this is essentially an implementation _plan_, with a quick test implementation of one element where relevant, not the implementation itself.

**Comment by rub1e** [comment]
> Add expenses in a foreign currency

We can't do any multi-currency stuff until we upgrade Xero to a Premium account.

If we think that's worth doing, great - if not, I'll draft a process for non-GBP expenses

**Comment by iteles** [comment]
@rub1e Am I right in thinking that that would be an additional £5.50 per month, £66 a year?

If so, let's upgrade, it will cost us more than that in dwyler time to do the conversions ourselves for dwyler summer receipts alone!

**Comment by rub1e** [comment]
I've had a quick look at the Xero `Projects` add-on and it's certainly worth looking into properly 👁  - so I'll start the free trial next week and see if it's worth the extra cost 💸 

The main functionality it adds is the ability to add invoices to a specific project, which lets you see the project's profitability at a glance, so useful for #373 

There are also a few other features like time tracking ⌚️  and costing which may or may not prove useful

**Comment by rub1e** [comment]
I've now had a play around with the Xero `Projects` product - it's between *quite* and *somewhat* useful.

If you want a general intro, you could do worse than watching this [five-minute video](https://tv.xero.com/detail/video/5755852909001/managing-projects) (try to ignore her saying `pr-oh-jects` the whole time)

The main points of interest are

* Visibility across all jobs at once - this is something we seriously lack atm
* Tracking `estimate vs actual` budgets
* Reporting (screenshot)

![image](https://user-images.githubusercontent.com/11595920/45038884-17582d00-b05a-11e8-8133-1855c60331e9.png)
* Timesheet-style functionality (add hours actually worked and see whether you've gone over/under allocation)
* Tracking actual staff cost/hr and incorporating that into project profitability (which is calculated automatically and visible at-a-glance)
* Quotes / costings
* More control over invoicing (e.g. with up-front deposits) and bills (can be assigned directly to a project)

The main drawbacks are:

* Learning/setup curve - it'll take effort to set up all the inventory items, play with reports etc (as ever with tools like this, it would take a little while before we're able to use it to its fullest)
* Would probably need to give access to ALL dwylers (PAYE & freelancers) and pay £5/month for each of them (I'm waiting for clarification from Xero)
* Wouldn't actually change how we manage work - we'd still need a new Project for every sprint.
  * Which then means we can't easily check profitability across a *client* - as the report produced would be per *sprint*
* Increase in admin time - this wouldn't necessarily be a bad thing if it the system saves admin time in the long run; but it does feel like the kind of thing where you if you take your eyes off it for a moment, the whole thing  collapses - I think it would require constant updating. (Again, not intrinsically a bad thing, but I suspect this is one of those tools that constantly makes work for yourself if you use it properly)
* No Calendar functionality - we'd still have to use the `company calendar`

I think we should have a chat about whether the advantages outweigh the drawbacks, but ultimately I suspect we'd be better off with a lo-fi spreadsheet that does *exactly* what we need than changing the way we work to fit into this tool (and still having to do extra work to see things the way we want to)

(And I'm not just saying that because I'm the one who'd have to set it up... 😜) 

(This is still a WIP, will update when Xero have replied to my queries)

**Comment by rub1e** [comment]
After discussion with @iteles and responses from Xero:

Xero would indeed charge us for each person involved in Projects.

While an extra £50/m wouldn't be unreasonable if it saved time elsewhere, we'd have to do so much work to get an output from Projects in the format we want that it's just not worth it (specifically the ability to see across a whole client/project rather than just per sprint)

In short, Xero Projects doesn't provide an easy solution to #373, and we're probably better off rolling our own to meet our specific needs

**Comment by rub1e** [comment]
@iteles the second task above:

> Pull information out for year end accounts 

Do you just mean the usual balance sheet / P&L reports we need for HMRC & Accountants? If so then that point can be checked off and I'll document the process as we do #461 

If there's anything else, let me know and I'll investigate when I look at the pain points in #382

**Comment by iteles** [comment]
@rub1e That's exactly it 👍

**Comment by rub1e** [comment]
I've answered the final outstanding question about invoicing pain-points at length [here](https://github.com/dwyl/hq/issues/382#issuecomment-428198420)

As far as I can see we've now answered all the points in the issue description (or at least put some explanation of our thinking in the relevant issue), so this issue can be closed.

**Comment by rub1e** [comment]
I'm not sure where this lives, but I've just checked and Xero Expenses costs £2.50 a month per user, and it would just be me and @iteles using it - so I'm very inclined to say it'll save more in man-hours than the £5/mo cost

So I'd say go for it!
